Importing a file with a name that starts with the package name outside of the package path provides incorrect error message

With the following directory structure:

```
src/main.zig
src/game_api.zig
src/game/game.zig
```

build.zig snippets:
```
const exe = b.addExecutable("ion", "src/main.zig");
...
const game = b.addSharedLibrary("game", "src/game/game.zig", .unversioned);
```

If `game.zig` has `const game_api = @import("../game_api.zig");`, then the resulting error is:

```
$ zig build
C:\Users\kcbanner\projects\ion\src\game\game.zig:4:26: error: unable to load 'C:\\Users\\kcbanner\\projects\\ion\\src\\game\api.zig': FileNotFound
const game_api = @import("../game_api.zig");
```

Not how it's trying to import `game\api.zig`.

However, if `game_api.zig` is changed to say, `foo_api.zig`, we get the correct message:
```
$ zig build
.\game\game.zig:5:18: error: import of file outside package path: '../foo_api.zig'
const game_api = @import("../foo_api.zig");
```

I believe the issue is that in `Module.zig:importFile`:

```
    if (!mem.startsWith(u8, resolved_path, resolved_root_path)) {
        return error.ImportOutsidePkgPath;
    }
```

Here `resolved_root_path` is `"C:\\Users\\kcbanner\\projects\\ion\\src\\game"`, so it matches with the `resolved_path` of `"C:\\Users\\kcbanner\\projects\\ion\\src\\game_api.zig"`. I think the solution might be to add the path separator to the end of `resolved_root_path`? 

I can make a PR for this, if that is an acceptable solution.

Thanks for looking into this!

Example repo is [here](https://github.com/kcbanner/zig-9537), currently I get:

```
$ zig build
C:\Users\kcbanner\projects\zig-9537\src\game\game.zig:1:26: error: unable to load 'C:\Users\kcbanner\projects\zig-9537\src\game\api.zig': FileNotFound
const game_api = @import("../game_api.zig");

$ zig version
0.9.1
```

--- a/src/Module.zig
+++ b/src/Module.zig
@@ -4367,7 +4367,7 @@ pub fn importFile(
     if (!mem.startsWith(u8, resolved_path, resolved_root_path) or
         // This prevents this check from triggering when the name of the
         // imported file starts with the root path's directory name.
-        mem.indexOfAny(u8, &.{resolved_path[resolved_root_path.len]}, "/\\") == null)
+        std.fs.path.isSep(resolved_path[resolved_root_path.len]))
     {
         return error.ImportOutsidePkgPath;
     }

--- a/src/Module.zig
+++ b/src/Module.zig
@@ -4364,7 +4364,11 @@ pub fn importFile(
     const resolved_root_path = try std.fs.path.resolve(gpa, &[_][]const u8{cur_pkg_dir_path});
     defer gpa.free(resolved_root_path);
 
-    if (!mem.startsWith(u8, resolved_path, resolved_root_path)) {
+    if (!mem.startsWith(u8, resolved_path, resolved_root_path) or
+        // This prevents this check from triggering when the name of the
+        // imported file starts with the root path's directory name.
+        mem.indexOfAny(u8, &.{resolved_path[resolved_root_path.len]}, "/\\") == null)
+    {
         return error.ImportOutsidePkgPath;
     }
     // +1 for the directory separator here.

--- a/src/Module.zig
+++ b/src/Module.zig
@@ -4367,7 +4367,7 @@ pub fn importFile(
     if (!mem.startsWith(u8, resolved_path, resolved_root_path) or
         // This prevents this check from triggering when the name of the
         // imported file starts with the root path's directory name.
-        std.fs.path.isSep(resolved_path[resolved_root_path.len]))
+        !std.fs.path.isSep(resolved_path[resolved_root_path.len]))
     {
         return error.ImportOutsidePkgPath;
     }

